Services/DataBaseService/Utils/MigrationEngine.cs

```diff
@@ -0,0 +1,269 @@
+        public void Migrate()
+        {
+            var connectionStringInitial = configuration.GetConnectionString("InitialString");
+            var connectionStringTradingStation = configuration.GetConnectionString("TradingStationString");
+            var scriptsLocation = configuration.GetSection("Locations")["MigrationScripts"];
+            var allScripts = Directory.GetFiles(scriptsLocation, "*.sql");
+            var scriptsToWriteDown = new Dictionary<string, ExecutedScript>
+            {
+                { "CreateDatabase", CreateDatabase(connectionStringInitial) },
+                { "CreateScriptsTable", CreateScriptsTable(connectionStringTradingStation) }
+            };
+            List<string> scriptsToExecute = FilterScriptsToExecute(connectionStringTradingStation, allScripts);
+            var lastScriptToExecute = "no script selected";
+
+            try
+            {
+                using (var conn = new SqlConnection(connectionStringTradingStation))
+                {
+                    conn.Open();
+                    SqlTransaction executingTransaction = conn.BeginTransaction();
+                    try
+                    {
+                        foreach (var fileName in scriptsToExecute)
+                        {
+                            lastScriptToExecute = fileName;
+                            var scriptCode = File.ReadAllText(fileName);
+                            using (var command = new SqlCommand(scriptCode, conn, executingTransaction))
+                                command.ExecuteNonQuery();
+                            scriptsToWriteDown.Add(fileName, new ExecutedScript(DateTime.Now, scriptCode));
+                        }
+                        executingTransaction.Commit();
+                        WriteDownExecutedScripts(connectionStringTradingStation, scriptsToWriteDown);
+                        Console.WriteLine("\tExecuted scripts written down.");
+                    }
+                    catch (SqlException e)
+                    {
+                        // TODO replace with logs
+                        Console.WriteLine(e.Message +
+                            $"\n\tError in the [{lastScriptToExecute}] script, or the connection is broken.");
+                        try
+                        {
+                            executingTransaction.Rollback();
+                            // TODO replace with logs
+                            Console.WriteLine("\tExecuting rollback is successful.");
+                        }
+                        catch
+                        {
+                            // TODO replace with logs
+                            Console.WriteLine("\tCouldn't rollback executing.");
+                            WriteDownExecutedScripts(connectionStringTradingStation, scriptsToWriteDown);
+                            Console.WriteLine("\tExecuted scripts were written down.");
+                        }
+                        throw;
```

Comment:
What for?

Reply:
Because when migration fails, we don't want to continue starting the app, do we?